src/expr/src/relation/func.rs

```diff
@@ -961,12 +961,15 @@ fn generate_series_ts(
     step: Interval,
     conv: fn(NaiveDateTime) -> Datum<'static>,
 ) -> Result<impl Iterator<Item = (Row, Diff)>, EvalError> {
-    if step.months == 0 && step.duration == 0 {
+    let normalized_step = i64::from(step.months) * 30 * 24 * 60 * 60 * 1_000_000
```

Comment:
I created the `as_microsecond()` method to clean this up. Instead of returning an `Option` or `Result` I just had it return an `i128`. I don't think we want an overflow to cause these checks to fail, since we only really use `normalized step` to check the direction and equality to 0.